src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
> so I feel like moving it up to help this scenario would be beneficial.

I don't agree.  This code is used when unzipping files larger than 2GB-ish.  It should be relatively rare to unzip files that large, and yet the moment a process does, it's going to end up holding onto a 2GB array for the lifetime of the process.  If it's a 10GB zip file, we'll end up holding onto 5 2GB-ish arrays for the lifetime of the process.

This doesn't seem like a good trade-off to me.  If we had built-in support for reclaiming arrays in the pool after some period of time, then maybe, but as it currently stands, the pool is effectively a memory leak.